src/tvdb/client.py

```diff
@@ -0,0 +1,229 @@
+from abc import ABC, abstractmethod
+from typing import ClassVar, Literal, final, overload, override
+
+import aiohttp
+from yarl import URL
+
+from src.settings import TVDB_API_KEY
+from src.tvdb.generated_models import (
+    MovieBaseRecord,
+    MovieExtendedRecord,
+    SearchResult,
+    SeriesBaseRecord,
+    SeriesExtendedRecord,
+)
+from src.tvdb.models import (
+    MovieExtendedResponse,
+    MovieResponse,
+    SearchResponse,
+    SeriesExtendedResponse,
+    SeriesResponse,
+)
+from src.utils.log import get_logger
+
+log = get_logger(__name__)
+
+type JSON_DATA = dict[str, JSON_DATA] | list[JSON_DATA] | str | int | float | bool | None  # noice
+
+type SeriesRecord = SeriesBaseRecord | SeriesExtendedRecord
+type MovieRecord = MovieBaseRecord | MovieExtendedRecord
+type AnyRecord = SeriesRecord | MovieRecord
+
+
+def parse_media_id(media_id: int | str) -> int:
+    """Parse the media ID from a string."""
+    return int(str(media_id).removeprefix("movie-").removeprefix("series-"))
+
+
+class _Media(ABC):
+    def __init__(self, client: "TvdbClient", data: SeriesRecord | MovieRecord | SearchResult):
+        self.data = data
+        self.client = client
+        self.name: str | None = self.data.name
+        self.overview: str | None = None
+        # if the class name is "Movie" or "Series"
+        self.entity_type: Literal["Movie", "Series"] = self.__class__.__name__  # pyright: ignore [reportAttributeAccessIssue]
+        if hasattr(self.data, "overview"):
+            self.overview = self.data.overview  # pyright: ignore [reportAttributeAccessIssue]
+        self.slug: str | None = None
+        if hasattr(self.data, "slug"):
+            self.slug = self.data.slug
+        if hasattr(self.data, "id"):
+            self.id = self.data.id
```

Comment:
So, if `id` isn't in `data`, it just won't be set at all? So we'll get an AttributeError on attempting to access it? Is that intentional? If not, consider allowing `id` to be `None`.

Reply:
I changed the pydantic models in the meantime. It was because there they were set to be nullable because of that dumb bug.